src/librbd/io/ImageRequestWQ.cc

```diff
@@ -535,9 +554,15 @@ void ImageRequestWQ<I>::process(ImageRequest<I> *req) {
                  << "req=" << req << dendl;
 
   req->send();
+  bool is_write = req->is_write_op();
+  if (is_write && m_image_ctx.max_write_iops) {
```

Comment:
Nit: you can create an inline helper method that returns a pointer to the struct when provided a boolean ```write_op``` parameter -- then this logic can be reused below.